internal/platform.h

```diff
@@ -0,0 +1,84 @@
+// Copyright 2015 The Gemmlowp Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// internal/platform.h: a place to put platform specific code
+
+#ifndef GEMMLOWP_INTERNAL_PLATFORM_H_
+#define GEMMLOWP_INTERNAL_PLATFORM_H_
+
+
+#ifdef _WIN32
+#include <windows.h>
+#include <malloc.h>
+#else
+#include <unistd.h>
+#endif
+
+namespace gemmlowp {
+  
+#ifdef _WIN32
+inline int posix_memalign(void **memptr, size_t alignment, size_t size) {
+  *memptr = _aligned_malloc(size, alignment);
+  return (*memptr == NULL);
+}
+
+inline void aligned_free(void *memptr) {
+  _aligned_free(memptr);
+}
+
+inline int ResolveMaxThreads(int max_threads) {
+  if (max_threads == 0) {
+    SYSTEM_INFO sysinfo;
+    GetSystemInfo(&sysinfo);
+    return sysinfo.dwNumberOfProcessors;
+  }
+  return max_threads;
+}
+
+inline double time() {
```

Comment:
time() was actually a pretty bad function name, it was barely good enough for benchmark code, but not ready to be promoted to a platform.h file.

If my recollection is correct that this is only used in the benchmark then just polyfill it here. Otherwise, rename it, perhaps to real_time_in_seconds.